UsersApp/AppDelegate/AppDelegate.swift

```diff
@@ -1,30 +1,69 @@
 import UIKit
-
+import GoogleSignIn
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
+    
     var window: UIWindow?
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         
         NetworkMonitor.shared.initiateConnectivityTracking()
         
+        GIDSignIn.sharedInstance.restorePreviousSignIn { user, error in
+            if let user = user {
+                print("Successfully restored previous sign-in for user: \(user.profile?.email ?? "")")
+                self.showSignedInState(for: user)
+            } else if let error = error {
+                print("Error restoring previous sign-in: \(error.localizedDescription)")
+                self.showSignedOutState()
+            } else {
+                print("No previous user found")
+                self.showSignedOutState()
+            }
+        }
         return true
     }
-
+    
+    func application(
+        _ app: UIApplication,
+        open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]
+    ) -> Bool {
+        var handled: Bool
+        
+        handled = GIDSignIn.sharedInstance.handle(url)
+        if handled {
+            return true
+        }
+        
+        return false
```

Comment:
why do you need the GIDSignIn to handle all URLs?
also, from the code above it makes sense to just `return handled`